tests/test_api.py

```diff
@@ -725,6 +725,12 @@ def test_targetfile_from_data(self) -> None:
         targetfile_from_data = TargetFile.from_data(target_file_path, data)
         targetfile_from_data.verify_length_and_hashes(data)
 
+    def test_targetfile_hash_prefix_paths(self) -> None:
+        target = TargetFile(
+            100, {"sha256": "abc", "md5": "def"}, "public/path/file.ext"
+        )
```

Comment:
for completeness maybe also:
* path without separator "file.ext"
* path without filename "path/.ext"
* empty path ""
* empty filename "path/"
* (add other path shenanigans if you can think of any more)

Reply:
Oh since its likely not obvious: empty path and empty filename probably don't make sense: please still test that the code does not e.g. throw exceptions if that happens